### PR TITLE
[#182563382] - Change e2e container to not run as root

### PR DESCRIPTION
**Pull request by dna-minn**

### CODE REVIEW

File: e2e-tests/pom.xml [hunk 1]
```diff
@@ -153,73 +153,48 @@
...
+							<artifactId>dd-javac-plugin</artifactId>
+							<version>0.1.7</version>
+							<repositoryType>PLUGIN</repositoryType>
+						</dynamicDependency>
+						<dynamicDependency>
+							<groupId>com.datadoghq</groupId>
+							<artifactId>dd-javac-plugin-client</artifactId>
+							<version>0.1.7</version>
+							<repositoryType>PLUGIN</repositoryType>
+						</dynamicDependency>
```
Comment (sharvari-kapadia) [inline e2e-tests/pom.xml:196]: @dna-minn Would you mind providing some context on these two `dynamicDependency` blocks being added here?

**Review by sharvari-kapadia — COMMENTED** [review]
@prajjwolmondal would you or someone on the publisher team mind looking at this PR since Minn has refactored the test folders so that they get picked up?

**Review by mbarkley — CHANGES_REQUESTED** [review]
@dna-minn requesting changes on this one so we don't forget to apply the fixes to the problems we learned about yesterday before merging this.

**Review by mbarkley — APPROVED** [review]
LGTM.

@prajjwolmondal we had to move all the e2e tests from `src/main/java` to `src/test/java` for this new simplified setup to work. Can you or someone else on the publisher team review to confirm that's okay?

**Review by usanthan — APPROVED** [review]

**Pull request merged**